examples/advanced/example_model_optimization.py

```diff
@@ -277,16 +279,24 @@ def enc_to_repr(self, enc: torch.Tensor) -> torch.Tensor:
 #   If you do not specify an ``optimizer``, the
 #   :func:`~pystiche.optim.default_model_optimizer`, i.e. :class:`~torch.optim.Adam` is
 #   used.
+#
+# .. note::
+#
+#   The weights of the provided transformer were trained with the
+#   `2014 training images <http://images.cocodataset.org/zips/train2014.zip>`_ of the
+#   `COCO dataset <https://cocodataset.org/>`_. The training was performed for
+#   ``num_epochs=2`` and ``batch_size=4``. Each image was center-cropped to
+#   ``256 x 256`` pixels.
 
 
-def train(
-    transformer, dataset, batch_size=4, epochs=2,
-):
-    if dataset is None:
-        raise RuntimeError(
-            "You forgot to define a dataset. For example, "
-            "you can use any image dataset from torchvision.datasets."
-        )
+def train(transformer, root, batch_size=4, epochs=2, image_size=256):
```

Comment:
Maybe we can remove the defaults and the values to the call? That would link this call to the note above.

```suggestion
def train(*, transformer, root, batch_size, epochs, image_size):
```

Reply:
Sure, thats fine. Is it a style conform thing?

We would also have to modify the training call later to include them:
`transformer = train(transformer, root, batch_size=4, epochs=2, image_size=256)`

Reply:
> Sure, thats fine. Is it a style conform thing?

The note that explains the default parameters for training was moved below this block. From a narrative perspective it would thus be off to have them here as default values.



> We would also have to modify the training call later to include them:
> `transformer = train(transformer, root, batch_size=4, epochs=2, image_size=256)`

Yes, we need to modify, but the call should look like:

```py
train(transformer=transformer, root=root, batch_size=4, epochs=2, image_size=256)
```

Due to the `*` in the beginning of the signature, all parameters are keyword only.